code/game/machinery/machinery.dm

```diff
@@ -464,10 +464,8 @@ Class Procs:
 /obj/machinery/tesla_act(var/power)
 	..()
 	if(prob(85))
-		emp_act(2)
+		explosion(src.loc,1,2,4,flame_range = 2)
```

Comment:
even with smoke it's not even close to as laggy as singulo was before lag removal
without smoke it will be even less
if that's still too bad we can always lower the explosion prob